src/sentry/snuba/discover.py

```diff
@@ -0,0 +1,230 @@
+        elif isinstance(aggregation[1], (set, tuple, list)):
+            aggregation[1] = [resolve_column(col) for col in aggregation[1]]
+    resolved["aggregations"] = aggregations
+
+    conditions = resolved.get("conditions")
+    if conditions:
+        for (i, condition) in enumerate(conditions):
+            replacement = resolve_condition(condition)
+            conditions[i] = replacement
+        resolved["conditions"] = list(filter(None, conditions))
+
+    # TODO add support for extracting having conditions.
+
+    orderby = resolved.get("orderby")
+    if orderby:
+        orderby = orderby if isinstance(orderby, (list, tuple)) else [orderby]
+        resolved_orderby = []
+
+        for field_with_order in orderby:
+            field = field_with_order.lstrip("-")
+            resolved_orderby.append(
+                u"{}{}".format(
+                    "-" if field_with_order.startswith("-") else "",
+                    field if field in derived_columns else resolve_column(field),
+                )
+            )
+        resolved["orderby"] = resolved_orderby
+    return resolved
+
+
+def transform_results(result, translated_columns):
+    # Translate back columns that were converted to snuba format
+    for col in result["meta"]:
+        col["name"] = translated_columns.get(col["name"], col["name"])
+
+    def get_row(row):
+        return {translated_columns.get(key, key): value for key, value in row.items()}
+
+    if len(translated_columns):
+        result["data"] = [get_row(row) for row in result["data"]]
+
+    # TODO(mark) add zerofill for timeseries data.
+    return result
+
+
+def query(selected_columns, query, params, orderby=None, referrer=None):
+    """
+    High-level API for doing arbitrary user queries against events.
+
+    This function operates on the public event schema and
+    virtual fields/aggregate functions for selected columns and
+    conditions are supported through this function.
+
+    The resulting list will have all internal field names mapped
+    back into their public schema names.
+
+    selected_columns (Sequence[str]) List of public aliases to fetch.
+    query (str) Filter query string to create conditions from.
+    params (Dict[str, str]) Filtering parameters with start, end, project_id, environment
+    orderby (str|Sequence[str]) The field to order results by.
+    """
+    snuba_filter = get_filter(query, params)
+
+    # TODO(mark) Refactor the need for this translation shim once all of
+    # discover is using this module
+    snuba_args = {
+        "start": snuba_filter.start,
+        "end": snuba_filter.end,
+        "conditions": snuba_filter.conditions,
+        "filter_keys": snuba_filter.filter_keys,
+        "orderby": orderby,
+    }
+    snuba_args.update(resolve_field_list(selected_columns, snuba_args))
+
+    # Resolve the public aliases into the discover dataset names.
+    snuba_args = resolve_discover_aliases(snuba_args)
```

Comment:
nit: Why not `snuba_args, translated_columns = resolved_discover_aliases(snuba_args)`. I think it's a bit weird to put the translated columns in the snuba_args dict since it's not an arg for Snuba

Reply:
That is a much better idea :+1: